datastore/src/main/java/org/jboss/pnc/datastore/repositories/ArtifactRepositoryImpl.java

```diff
@@ -69,4 +70,19 @@ public Set<Artifact> withIdentifierAndSha256(Collection<Artifact.IdentifierSha25
         return artifacts;
     }
 
+    @Override
+    public List<Artifact> withSha256AndDependantBuildRecordIdIn(String sha256, Set<Base32LongID> buildRecordIds) {
+        return queryWithPredicates(ArtifactPredicates.withSha256AndDependantBuildRecordIdIn(sha256, buildRecordIds));
+    }
+
+    @Override
+    public List<Artifact> withIdentifierAndSha256(String identifier, String sha256) {
+        return queryWithPredicates(ArtifactPredicates.withIdentifierAndSha256(identifier, sha256));
+    }
+
+    @Override
+    public List<Artifact> withSha256In(Set<String> sha256) {
+        return queryWithPredicates(ArtifactPredicates.withSha256In(sha256));
+    }
```

Comment:
Why are you adding these as explicit methods, when ther are Predicates?

Reply:
Because it might be me not good enough (possible!) with mocks, but I could not find a way to mock this behavior:
when artifactRepository.queryWithPredicates is called with the Predicate "ArtifactPredicates.withIds" then return artifacts, otherwise if  artifactRepository.queryWithPredicates is called with the Predicates "ArtifactPredicates.withSha256AndDependantBuildRecordIdIn", "ArtifactPredicates.withIdentifierAndSha256", "ArtifactPredicates.withSha256In" then return an empty list.... this is for the test below where you found already an improvement.

Reply:
Ah ok. It's fine then I think.

What you describe, i think i implemented something like that by something like this:
```
MyObject specificObejct = new MyObject(); // maybe mock(MyObject.class)
mock(ArtifactPredicates.withSha256AndDependantBuildRecordIdIn()).thenReturn(specificObject);
mock(artifactRepository.queryWithPredicates(specificObject)).thenReturn(whatIwant);
```
However I'm not sure from top of my head how/if mocking static methods is done/possible

Reply:
I will look again but I have tried endless ways, above one probably included (too many to remember!) :-)

Reply:
nope.... as soon as I mock more Predicates I get the `org.mockito.exceptions.misusing.InvalidUseOfMatchersException` because the mock cannot distinguish the different predicates (and their different signature). So explicit methods are the only way I am afraid.